### ISSUE TITLE
Generated 'User' entity's 'password' field's column is nullable

### ISSUE THREAD

**Issue by yatendragoel** [jhipster/generator-jhipster#7586]
##### **Overview of the issue**

Generated `User.java` entity's `password` field looks like below:

    @JsonIgnore
    @NotNull
    @Size(min = 60, max = 60)
    @Column(name = "password_hash", length = 60)
    private String password;

Since, it is `@NotNull` at the field level, shouldn't it be `nullable = false` at the column level as well?

    @Column(name = "password_hash", length = 60, nullable = false)

**Comment by jdubois** [comment]
We have some use cases (like login with Google) where there is no password, that's why the column is `nullable`, but indeed that should be refined depending on your use case - please follow our guidelines next time, so we know your configuration and what you are trying to do

**Comment by yatendragoel** [comment]
But then `@NotNull` should also be removed from the auto-generated code. The whole point I want to highlight is that nullability of a field should match with the nullability of the column. Either both should be allowed to have `null` values or none of them should be allowed.

**Comment by jdubois** [comment]
Yes @yatendragoel - sorry that's what I meant by "refined" (not sure this is good English!), we should have a check here, and it should match the same for both the entity and the database column
(I haven't looked at the code yet, but I don't expect this to be very complex)

**Comment by yatendragoel** [comment]
Cool. So we can fix this and this is a valid bug. I am letting this issue opened so that it can be picked up for the fix.

Maybe I will only pick this up to fix depending on my availability in the coming days.

**Comment by deepu105** [comment]
Yes if you can do a PR that would be great

**Comment by thainan10** [comment]
I did a PR to solve it.
Not quite sure if it's enough but I think so.
As mentioned and I checked, the liquibase changelog already creates a nullable password field.

**Comment by jdubois** [comment]
Sorry @thainan10 I thought you wanted the database column to be non-nullable, not the Java field to be nullable.

**Comment by thainan10** [comment]
That's ok. In this case, I have to change the liquibase script to make the password field non-nullable.
I'll do it and submit a PR. That's ok?
